example/flaskr/ingredient.py

```diff
@@ -0,0 +1,111 @@
+import functools
+
+from flask import Blueprint, json
+from flask import flash
+from flask import g
+from flask import redirect
+from flask import render_template
+from flask import request
+from flask import session
+from flask import url_for
+from flaskr.db import get_db
+
+bp = Blueprint("ingredient", __name__, url_prefix="/ingredient")
+RATINGS = [ 'r_nourishment',
+            'r_value',
+            'r_human_welfare',
+            'r_animal_welfare',
+            'r_resource_cons',
+            'r_biodiversity',
+            'r_global_warming',]
+
+@bp.route("/<str:ingredientName>", methods=("GET", "POST"))
+def getIngredient():
+    """Retrieve an Ingredient by name
+    Returns None if Ingredient not present
+    """
+    if request.method == "GET":
+        db = get_db()
+        error = None
+
+        if not ingredientName:
+            error = "Ingredient name is required."
+        else:
+            ingredient = db.execute(
+                "SELECT * FROM ingredient WHERE name = ?", ingredientName)
+                .fetchone()
+        
+        if ingredient is None:
+            abort(404, f"Ingredient with name {ingredientName} doesn't exist.")
+        
+        return ingredient
+
+
+@bp.route("/<str:ingredientName>/alt", methods=("GET"))
```

Comment:
I think if a request is made to a route which does not allow any other routes (i.e. POST, DELETE, etc), it should get rejected with a 405 automatically: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405

So I don't think you need to include the `methods` parameter here, as well as encompassing everything in the if statement (same with the other routes in this file).